src/components/__tests__/IssueSearchAdvanced.spec.ts

```diff
@@ -32,4 +32,32 @@ describe('IssueSearchAdvanced', () => {
   it('should render', () => {
     expect(wrapper.exists()).toBe(true);
   });
+
+  it('should show advanced search', async () => {
+    const content = wrapper.find('details');
+    expect(content.exists()).toBe(true);
+  });
+
+  it('should emit on search button click', async () => {
+    const searchButton = wrapper.find('button[type="submit"]');
+    expect(searchButton.exists()).toBeTruthy();
+    await searchButton.trigger('submit');
+    await flushPromises();
+    const setFilterEvent = wrapper.emitted('set:filters');
+    expect(setFilterEvent[0][0]).toEqual({});
+  });
+
+  it('should emit with filters on search button click', async () => {
+    const selectDropdown = wrapper.find('select.form-select.search-facet-field');
+    await selectDropdown.setValue(selectDropdown.findAll('option')[1].element.value);
+    await selectDropdown.trigger('change');
+    const inputField = wrapper.find('input.form-control');
+    await inputField.setValue('test'); 
+    const searchButton = wrapper.find('button[type="submit"]');
+    expect(searchButton.exists()).toBeTruthy();
+    await searchButton.trigger('submit');
+    await flushPromises();
+    const setFilterEvent = wrapper.emitted('set:filters');
+    expect(setFilterEvent[0][0]).toEqual({ acknowledgments__name: 'test' });
```

Comment:
I like the thoroughness, but ultimately we may want to test the external behavior of the network request, so we're not testing internal implementation. For example, if the `set:filters` event is renamed this test will break although the functionality could be correct.

Instead we might want to spy/intercept this network request here

https://github.com/RedHatProductSecurity/osim/blob/b38a6274cddf3fb85a1bbffec320bfed6a9a039e/src/components/IssueSearchAdvanced.vue#L136

This spy setup might look something like

```ts
import { http, HttpResponse } from 'msw';
import { setupServer } from 'msw/node';


const getParamsHandler = http.get('/osidb/api/v1/flaws', async ({ request }) => {
  const reader = request.params?.getReader();

  if (!reader) {
    throw new Error('Request body is not available');
  }

  const result = await reader.read();

  if (result.done) {
    throw new Error('Request body stream is already read to the end');
  }

  const decoder = new TextDecoder('utf-8');
  const paramsDecoded = decoder.decode(result.value);
  return paramsDecoded;
});

const server = setupServer(getParamsHandler);

```
and include setup and teardown:
```ts
beforeAll(() => {
  server.listen({ onUnhandledRequest: 'error' });
}
// ...
afterEach(() => server.resetHandlers());
// ...
afterAll(() => server.close());
```
Then you'd make a function for the mocked request...
```ts
function mockedAdvancedSearch(params: Record<string, string>) {
  return axios({
    method: 'get',
    url: '/osidb/api/v1/flaws',
    params,
  })
    .then((response) => {
      // Not sure about this part
    })
    .catch(console.error);
}
```

```ts
// .. in test 'should emit with filters' for IssueSearchAdvanced.spec.ts
    const result = await mockedPutFlaw(flaw.uuid, flaw);
    expect(result.owner).toBe('networking test owner');
```

Reply:
I haven't run any of the code I've written here but it is adapted from https://github.com/RedHatProductSecurity/osim/blob/c46130106f74763ecbd8aa7314d82934d38f419f/src/components/__tests__/FlawForm.spec.ts

Reply:
I think since we are writing test for useFlaws and IssueSearchAdvanced. we don't need to mock api. just mock function in useFlaws. So IssueSearchAdvanced.spec.ts file, we just test that emit is called or not.
In useFlaws.spec.ts, we are using getFlaws from flawservice. i think we can mock that getFlaws instead of mocking api. what do you think? @superbuggy

Reply:
Ah yeah that makes way more sense